src/main/java/com/google/cloud/tools/gradle/endpoints/client/EndpointsClientPlugin.java

```diff
@@ -0,0 +1,183 @@
+/*
+ *  Copyright (c) 2016 Google Inc. All Right Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.gradle.endpoints.client;
+
+import com.google.cloud.tools.gradle.endpoints.client.task.GenerateClientLibrariesTask;
+import com.google.cloud.tools.gradle.endpoints.client.task.GenerateClientLibrarySourceTask;
+import com.google.cloud.tools.gradle.endpoints.client.task.GetDiscoveryDocsFromDependenciesTask;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Plugin definition for Endpoints Clients (will generate source when applied)
+ *
+ * All tasks are hidden. Configuration of source discovery docs is from two ways:
```

Comment:
could there be use-cases where people might want a bit more control about how this works?  I like the automated nature of this, but I'm wondering if it's flexible enough.